### DISCUSSION TITLE
Default Query Function for useMutation

### DISCUSSION THREAD

**Discussion by apenab** [TanStack/query#1234] [Q&A]
Is there some way of get [this](https://react-query-beta.tanstack.com/guides/default-query-function) behavior, but for `useMutation`

**Comment by richraid21** [comment]
I'd like to bring this question back to ask if this choice to not support a default query function for `useMutation` was an intentional omission due to an architecture limitation or does it just need to be implemented? If the latter, I'd like to add it.

**Comment by TkDodo** [comment]
I think this should work just like with `useQuery`. `useMutation` also takes the object syntax:

```
useMutation({ mutationFn: () => ... })
```

so if you have a `mutationFn` defined in the `defaultOptions` for mutation on the `queryClient`, it should be picked up. can you provide an example that shows what is not working?
